assets/scripts/characters/mario/MarioStateMachine.ts

```diff
@@ -0,0 +1,162 @@
+// Learn TypeScript:
+//  - https://docs.cocos.com/creator/manual/en/scripting/typescript.html
+// Learn Attribute:
+//  - https://docs.cocos.com/creator/manual/en/scripting/reference/attributes.html
+// Learn life-cycle callbacks:
+//  - https://docs.cocos.com/creator/manual/en/scripting/life-cycle-callbacks.html
+
+import StateMachine from '../../../utils/StateMachine';
+import Mario from './Mario';
+import MARIO_STATES from './MARIO_STATES';
+import MOVEMENT_TYPE from './MOVEMENT_TYPE';
+
+const { ccclass } = cc._decorator;
+
+@ccclass
+export default class MarioStateMachine extends StateMachine<MARIO_STATES, Mario> {
+  private deathAux = true;
+
+  public onAnimationStart(name: string): void {
+    if (name === 'death') {
+      if (this.deathAux) {
+        this.deathAux = false;
+
+        cc.tween(this.node)
+          .call(() => this.actor.animationComponent.pause())
+          .delay(0.75)
+          .call(() => {
+            this.actor.animationComponent.resume();
+            this.actor.getComponents(cc.Collider).forEach(element => {
+              // eslint-disable-next-line no-param-reassign
+              element.enabled = false;
+            });
+            this.actor.rigidBody.gravityScale = 0.3;
+            this.actor.rigidBody.linearVelocity = cc.v2(0, 300);
+          })
+          .delay(5)
+          .call(() => this.actor.node.destroy())
+          .start();
+      }
+    }
+  }
+
+  public onAnimationEnd(name: string): void {
+    if (name === 'grow_up') {
+      this.actor.canMove = true;
+    }
```

Comment:
The player don't stop to walk, all the game pauses when mario get a power

Reply:
This is a temporary measure, as there isn't a game controller yet. I plan to implement one in the next PR I'll work on.